src/Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs

```diff
@@ -202,18 +211,30 @@ public override Task SendGroupExceptAsync(string groupName, string methodName, o
             var group = _groups[groupName];
             if (group != null)
             {
-                var message = CreateInvocationMessage(methodName, args);
-                var tasks = group.Values.Where(connection => !excludedIds.Contains(connection.ConnectionId))
-                    .Select(c => c.WriteAsync(message).AsTask());
+                var message = CreateSerializedInvocationMessage(methodName, args);
+                var tasks = SendMessageToConnections(message, group.Values.Where(connection => !excludedIds.Contains(connection.ConnectionId)));
                 return Task.WhenAll(tasks);
             }
 
             return Task.CompletedTask;
         }
 
-        private SerializedHubMessage CreateInvocationMessage(string methodName, object[] args)
+        private IEnumerable<Task> SendMessageToConnections(SerializedHubMessage hubMessage, IEnumerable<HubConnectionContext> connectionContexts)
         {
-            return new SerializedHubMessage(new InvocationMessage(target: methodName, argumentBindingException: null, arguments: args));
+            foreach (var connectionContext in connectionContexts)
+            {
+                yield return connectionContext.WriteAsync(hubMessage).AsTask();
```

Comment:
naw, no yielding. This looks allocation unfriendly. We should be the code inline.